pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -271,36 +264,30 @@ type mgr struct {
 }
 
 func (m *mgr) createDefaultProject(obj runtime.Object) (runtime.Object, error) {
-	return m.createProject(project.Default, v32.ClusterConditionconditionDefaultProjectCreated, obj, defaultProjectLabels, defaultProjects)
+	return m.createProject(project.Default, v32.ClusterConditionconditionDefaultProjectCreated, obj, defaultProjectLabels)
 }
 
 func (m *mgr) createSystemProject(obj runtime.Object) (runtime.Object, error) {
-	return m.createProject(project.System, v32.ClusterConditionconditionSystemProjectCreated, obj, systemProjectLabels, systemProjects)
+	return m.createProject(project.System, v32.ClusterConditionconditionSystemProjectCreated, obj, systemProjectLabels)
 }
 
-func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object, labels labels.Set, projectMap map[string]bool) (runtime.Object, error) {
-	return cond.DoUntilTrue(obj, func() (runtime.Object, error) {
+func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object, labels labels.Set) (runtime.Object, error) {
+	return cond.Do(obj, func() (runtime.Object, error) {
```

Comment:
Dropping the `DoUntilTrue` makes the condition pointless since it's not used anywhere else, and it also protects from cache misses and needlessly hitting the API. 

Have we considered doing something similar to the token cleanup and doing a one time drop of the condition on upgrade?

Reply:
Yes. I considered the setting the condition(s) to `Unknown` when the `projectLifecycle` controller is created, but that would introduce error handling where there is currently none. You indicated in a previous change I made that introducing error handling where there is none close to a release is a bad idea. I personally prefer this approach.

Reply:
Not in controller creation, you are correct that's a bad idea. After rancher is running, similar to token cleanup i.e. https://github.com/rancher/rancher/pull/33849/files

Reply:
Or https://github.com/rancher/rancher/blob/release/v2.6/pkg/multiclustermanager/app.go#L220 depending on what you need to do this.